compiler/hash-ast/src/origin.rs

```diff
@@ -0,0 +1,48 @@
+//! Auxiliary data structures for denoting where common AST nodes are located
+//! and which context they are used in.
+
+use std::fmt::Display;
+
+/// Denotes the current block kind.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum BlockOrigin {
+    /// Root block of the file.
+    Root,
+    /// The block is a `mod` block.
+    Mod,
+    /// The block is from a `trait` definition.
+    Trait,
+    /// The block is an `impl` block.
+    Impl,
+    /// Block is a body of some kind.
+    Body,
+    /// A dummy [BlockOrigin] kind, only used for error
+    /// reporting and grouping `constant` blocks together.
+    Const,
+}
+
+impl BlockOrigin {
+    /// Convert the [BlockOrigin] into a string which can be used for displaying
+    /// within error messages.
+    fn to_str(self) -> &'static str {
```

Comment:
Why not directly impl Display?

Reply:
This method is being used in other contexts that don't demand `fmt::Display` to be implemented.